src/main.rs

```diff
@@ -194,36 +196,31 @@ impl MainWin {
     }
 
     fn file_open(&self, hwnd_owner: HWND) {
-        unsafe {
-            let mut pfd: *mut IFileDialog = null_mut();
-            let hr = ole32::CoCreateInstance(&uuid::CLSID_FileOpenDialog,
-                null_mut(),
-                winapi::CLSCTX_INPROC_SERVER,
-                &uuid::IID_IFileOpenDialog,
-                &mut pfd as *mut *mut winapi::IFileDialog as *mut winapi::LPVOID
-                );
-            if hr != winapi::S_OK {
-                return;  // TODO: should be error result
-            }
-            (*pfd).Show(hwnd_owner);
-            let mut result: *mut winapi::IShellItem = null_mut();
-            (*pfd).GetResult(&mut result);
-            if !result.is_null() {
-                let mut display_name: LPWSTR = null_mut();
-                (*result).GetDisplayName(SIGDN_FILESYSPATH, &mut display_name);
-                if let Some(filename) = display_name.from_wide() {
-                    // Note: this whole protocol has changed a lot since the
-                    // 0.2 version of xi-core.
-                    self.send_edit_cmd("open", &json!({
-                        "filename": filename,
-                    }));
-                }
-                ole32::CoTaskMemFree(display_name as LPVOID);
-                (*result).Release();
-            } else {
-                //println!("result is null");
-            }
-            (*pfd).Release();
+        let filename = unsafe { get_open_file_dialog_path(hwnd_owner) };
+        if let Some(filename) = filename {
+            // TODO: Do not save here and pull from Xi core based on `self.state.view_id` instead?
```

Comment:
See my review comment in `struct MainWinState` above, I will remove this line in the next commit.